### PR TITLE
Wait for another reconcile request to fix flaky test in CI

### PR DESCRIPTION
**Pull request by thelabdude**
Fixes: #196

### CODE REVIEW

File: controllers/solrprometheusexporter_controller_test.go [hunk 1]
```diff
@@ -656,10 +657,14 @@ func TestMetricsReconcileWithUserProvidedConfig(t *testing.T) {
 	updatedConfigXml := "<config>updated by user</config>"
 	updateUserProvidedConfigMap(testClient, g, userProvidedConfigMapNN, map[string]string{util.PrometheusExporterConfigMapKey: updatedConfigXml})
 
-	// reconcile should happen again
+	// reconcile should happen again 2x
 	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedMetricsRequest)))
+	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedMetricsRequest)))
+
```
Comment (HoustonPutman) [inline controllers/solrprometheusexporter_controller_test.go:663]: In the [solrxml test](https://github.com/apache/lucene-solr-operator/blob/main/controllers/solrcloud_controller_test.go#L772), we added a 250 ms sleep to make sure it passed consistently. Might make sense to have here as well.

File: controllers/solrprometheusexporter_controller_test.go [hunk 2]
```diff
@@ -656,10 +657,14 @@ func TestMetricsReconcileWithUserProvidedConfig(t *testing.T) {
...
+	deployment = &appsv1.Deployment{}
+	err = testClient.Get(context.TODO(), metricsDKey, deployment)
+	g.Expect(err).NotTo(gomega.HaveOccurred())
 
-	deployment = expectDeployment(t, g, requests, expectedMetricsRequest, metricsDKey, userProvidedConfigMap.Name)
```
Comment (HoustonPutman) [inline controllers/solrprometheusexporter_controller_test.go:662]: Was `expectDeployment` not giving you the updated deployment?

Comment (thelabdude) [reply controllers/solrprometheusexporter_controller_test.go:662]: that was me just trying to simplify test logic but I changed it back

**Review by HoustonPutman — APPROVED** [review]
Looks good!

**Pull request merged**